Kernel/FileSystem/VirtualFileSystem.cpp

```diff
@@ -286,6 +286,76 @@ ErrorOr<void> VirtualFileSystem::unmount(Custody& mountpoint_custody)
     return unmount(guest_inode, custody_path->view());
 }
 
+Stack<Mount*, 16> VirtualFileSystem::get_mounts_stack()
+{
+    Stack<Mount*, 16> stack;
+    m_mounts.with([&](auto& mounts) -> void {
+        for (auto& mount : mounts) {
+            if (!stack.push(&mount))
+                break;
+        }
+    });
+    return stack;
+}
+
+ErrorOr<void> VirtualFileSystem::unmount_all(Badge<PowerManagementTask>)
```

Comment:
So... why is it ok to continue on with shutdown after the first filesystem fails to unmount? This code as written will not try to unmount every FS at least once. it always bails on the first error.

Reply:
Huh, I missed that. Definitely should try unmounting everything, since we’ve had bugs in the past with non-unmountable file systems.